### PR TITLE
## [2.0.0] - 2023-10-04

### PR DESCRIPTION
**Pull request by qemanuel**
### Changed

- s3 persistent storage replaced by ebs volume
- Inputs
- Launch configuration replaced by Launch template
- Docker version updated

### Added

- Useful outputs.
- Default values and types for most inputs.
- tags.
- Fixed version of the openvpn docker image.

### CODE REVIEW

**Review by Ciejo — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -10,36 +10,42 @@ Terraform module which creates an OpenVPN EC2 instance in AWS.
...
-| vpc\_id | VPC ID where OpenVPN will be deployed. | `string` | `""` | yes |
-| ami\_id | AMI ID to user for the OpenVPN EC2 instance. | `string` | `""` | yes |
-| instance\_type | OpenVPN EC2 instance type. | `string` | `""` | yes |
-| subnet\_ids | Public subnet ids from the designed VPC. | `list[string]` | `[]` | yes |
-| peered_networks | CIDRs blocks which OpenVPN will be able to route the traffic to/from. | `string` | `""` | yes |
-| docker_cidr | CIDR for Docker service. | `list[string]` | `[]` | yes |
-| compose_cidr | CIDR for `docker-compose.yaml`. | `list[string]` | `[]` | yes |
+| compose\_cidr | CIDR for default compose network. | `string` | `"192.168.127.1/24"` | no |
+| docker\_cidr | CIDR for default docker network. | `string` | `"192.168.125.1/24"` | no |
+| instance\_type | OpenVPN EC2 instance type. | `string` | `"t3.small"` | no |
```
Comment (ivours) [inline README.md:31]: Can we try with a `t3a.micro` as default?

File: resources/scripts/create_client.sh [hunk 2]
```diff
@@ -1,11 +1,11 @@
 
 if [ $(whoami) == "ubuntu" ] && [ -n "$1" ] ; then
     mkdir -p ./keys
-	docker-compose run --rm openvpn easyrsa build-client-full $1 nopass
-	docker-compose run --rm openvpn ovpn_getclient $1 > ./keys/$1.ovpn
+	docker compose run --rm openvpn easyrsa build-client-full $1 nopass
+	docker compose run --rm openvpn ovpn_getclient $1 > ./keys/$1.ovpn
 elif [ $(whoami) != "ubuntu" ]; then
-	echo "It must be run as ubuntu."
+	echo "This script must be ran as ubuntu."
```
Comment (ivours) [inline resources/scripts/create_client.sh:7]: The past participle of run is `run`, it's an irregular verb

File: resources/scripts/revoke_client.sh [hunk 3]
```diff
@@ -1,10 +1,10 @@
 if [ $(whoami) == "ubuntu" ] && [ -n "$1" ] ; then
-	sudo docker-compose run --rm openvpn ovpn_revokeclient $1 remove
+	docker compose run --rm openvpn ovpn_revokeclient $1 remove
 	echo "Client $1 has been removed."
 	rm ./keys/$1.ovpn
 elif [ $(whoami) != "ubuntu" ]; then
-	echo "It must be run as ubuntu."
+	echo "This script must be ran as ubuntu."
```
Comment (ivours) [inline resources/scripts/revoke_client.sh:6]: The past participle of run is run, it's an irregular verb

**Review by ivours — COMMENTED** [review]
Just to understand: Why aren't we using S3 as persistent storage anymore, and replacing it with EBS? The advantage of using S3 is that we have the persistent data always available in case of instance re-creation by the ASG. In case we use an EBS volume, the data would be lost on instance re-creation.

**Comment by qemanuel** [comment]
> Just to understand: Why aren't we using S3 as persistent storage anymore, and replacing it with EBS? The advantage of using S3 is that we have the persistent data always available in case of instance re-creation by the ASG. In case we use an EBS volume, the data would be lost on instance re-creation.

This EBS will persist too, it's created outside the Launch-template and it's attached to the instance in by the user-data. So, we have the same benefits (persistence) and we can take advantage of the EBS features which purpose is more suitable for file systems than the S3 buckets. For instance we could backing it up using snapshots. Also is easier to understand for somebody who didn't create the stack, because the same EC2 UI shows you the attached volume and so on.

**Pull request merged**